src/QueryMiddleware.php

```diff
@@ -143,4 +192,15 @@ private function validateRequestBody(array $body): void
             }
         }
     }
+
+    private function extractParamsFromRequest(ServerRequestInterface $request): array
+    {
+        $routeParams = $this->routeParamsExtractor->extractRouteParams($request);
+        $queryParams = $request->getQueryParams();
+
+        $payload = ['payload' => array_merge($routeParams, ['query' => $queryParams])];
```

Comment:
this would make. 'query' a reserved keyword for params coming from the route for queries as the GET vars would override those.